Address some of the issues with the security policy (#84)
openssl/general-policies: policies/security-policy.md

Reviewer: We also include other trusted individuals as part of the core team. Should we say that here?

Author: I've reorganized this section.

## Reporting security issues

If you wish to report a possible security issue in OpenSSL please
[notify us](/community/#reporting-security-bugssecurityreports) by
sending an e-mail to
[openssl-security@openssl.org](mailto:openssl-security@openssl.org).

If you have multiple issues to report, please always send a separate

> e-mail for each issue.

Reviewer 2: I disagree here; it doesn't matter how we get the report. It's our choice how we handle it; you can split it into multiple parts afterward.  
It must be as easy as possible for the reporter, with no artificial impediments.

Author: No, Anton. It does not mean we will completely ignore such reports but the current practice of reporting 20 issues of which barely one is worth a CVE in one mail skews the effort towards us unproportionally. I insist on keeping this here. It is a polite request to the reporter to spend some effort on their side.

Author: Also with such multi-reports the risk of overlooking one of the issues included in such report is definitely non-negligible and we want to avoid that.

Reviewer 2: okay

## Issue triage

Notifications are received by the OpenSSL Security Response Team ([SRT])
designated by the OpenSSL Foundation and the OpenSSL Corporation directors

> which contain resources from within the OpenSSL Foundation and the

Reviewer 2: ```suggestion
designated by the OpenSSL Foundation and the OpenSSL Corporation directors. The
SRT contains resources from within the OpenSSL Foundation and the
```

> OpenSSL Corporation and also members from the Committers community.

Reviewer 2: All members? Some members?

Author: Clarified

## Reporting security issues

If you wish to report a possible security issue in OpenSSL please
[notify us](/community/#reporting-security-bugssecurityreports) by

> an e-mail to [openssl-security@openssl.org](mailto:openssl-security@openssl.org).

Reviewer 3: sending an e-mail ?

If you have multiple issues to report, please always send a separate

> e-mail per each issue.

Reviewer: either "for each issue" or "per issue" but not "per each issue"

Author: Fixed

 - physical fault injection
 - physical observation side channels (e.g. power consumption,
   EM emissions, etc)

>  - those which just cause a denial of service of the OpenSSL application

Reviewer 3: OpenSSL application = OpenSSL command line utility?

Author: Done

 - API misuse by an application where such API is not supposed to be

>    directly exposed to an attacker-controlled data

Reviewer: ```suggestion
   directly exposed to attacker-controlled data
```

   configurations and which are also likely to be exploitable. Examples
   include significant disclosure of the contents of server memory
   (potentially revealing user details), vulnerabilities which can be easily
   exploited remotely to compromise server private keys or where remote code
   execution is considered likely in common situations. These issues will be
   kept private and will trigger a new release of all supported versions. We
   will attempt to address these as soon as possible.
 - <a name="high">**High**</a> Severity. This includes issues that are of a
   lower risk than critical, perhaps due to affecting less common
   configurations, affecting only protocol clients, or which are less likely
   to be exploitable. These issues will be kept private and will trigger
   a new release of all supported versions. We will attempt to keep the time
   these issues are private to a minimum; our aim would be no longer than
   a month where this is something under our control.
 - <a name="moderate">**Moderate**</a> Severity. This includes issues whose
   impact is only a denial of service, flaws in protocols that are not

>    commonly used (such as CMP), and local flaws. These will in general be kept

Reviewer: To describe CMP as a protocol that is "not commonly used" seems incorrect.

Author: We had DTLS as less commonly used protocol which is even less true IMO. I will change not -> less.

   private until the next release, and that release will be scheduled so
   that it can roll up several such flaws at one time.

>  - <a name="low">**Low**</a> Severity. This includes issues such as those

Reviewer 4: As you can probably guess, my vast preference here would be to swing the other way, and just say the lows will normally be fixed publicly but *may* handled with the release of a moderate. 

If the holdup is the CVE in the commit message, can we simply say that we will publish a list associating the CVE to the commit and provide full information in the next scheduled update release?

Author: I do not think we want to change the process to just fixing the issue silently and publishing it only with the release. That would be IMO a change in wrong direction. However the current process in the policy is already bad in this regard as it is partially doing that - the CVE id is properly published only after the fix is merged, etc.

So I am quite strongly standing behind my proposal.

   that only affect the openssl command line utility, cause only a denial
   of service for less common protocols or on client side only, or require
   multiple unlikely conditions to be fulfilled for the attack to succeed.
   These will in general be handled in the similar way as the **Moderate**

>    issues. However, as an exception, they might be developend and fixed

Reviewer 5: ? s/developend/developed ?

Author: Fixed

>    issues. However, as an exception, they might be developend and fixed

Reviewer 3: Not sure why the exception is needed here.
I would have thought it would either be classed as a CVE and hidden until release, or its not classed as a CVE and is done in public..

The only exception might be if someone puts up a public PR, and then it is decided its a CVE.

Author: We need such exception for issues like https://github.com/openssl/srt/issues/145 IMO.

   in public before the next release, for example in case the flaw is

>    rather theoretical and fixing it requires extensive design and

Reviewer 5: how about:

```
and the fix implementation is a non-trivial change.
```

Author: I prefer the current wording.

   development. In such case we will update the vulnerabilities page and

>    note the issue CVE in the changelog and commit message, but they may not

Reviewer 4: "but they may not to give our scheduled update release..." this is a bit confusing. who/what is "they"?

Author: reworded

 - <a name="low">**Low**</a> Severity. This includes issues such as those
   that only affect the openssl command line utility, cause only a denial
   of service for less common protocols or on client side only, or require
   multiple unlikely conditions to be fulfilled for the attack to succeed.
   These will in general be handled in the similar way as the **Moderate**
   issues. However, as an exception, they might be developed and fixed
   in public before the next release, for example in case the flaw is
   rather theoretical and fixing it requires extensive design and
   development. In such case we will update the vulnerabilities page and
   note the issue CVE in the changelog and commit message, but we will not
   publish new releases immediately.

## Prenotification policy

 - Where we are planning an update that fixes security issues we will notify
   the [openssl-announce list](https://groups.google.com/a/openssl.org/g/openssl-announce/)
   to give our scheduled update release date and time and the severity of
   issues being fixed by the update. No further information about the issues
   will be given. This is usually done one week before the actual release.

>  - Where we are planning an update that include **Critical**, **High**, or

Reviewer 2: ```suggestion
 - Where we are planning an update that includes **Critical**, **High**, or
```

   - We may withdraw notifying certain organisations from future
     prenotifications if they leak issues before they are public or over time
     do not add value.
   This private prenotification is usually done two weeks before the actual

>    release.

Reviewer: I'd put this sentence earlier (before the list of types of organisation)

Author: Done

   The most effective way for these sites to get protected is to get an
   updated version from that vendor.

## Security release update recommendations

Our security advisories describe the affected configurations or applications.

> We always recommend to review the advisories before rushing to update the

Reviewer: ```suggestion
We always recommend to review the advisories before updating
```

Author: Fixed

systems running with the unfixed releases.

However in general it is a good idea to update the systems as soon as possible
with the security update releases that contain **Critical** or **High**
severity issue fixes.

As the **Moderate** and **Low** severity issues are either unlikely to be
exploitable and/or the impact of the successful exploit is limited, we

> recommend to schedule the updates without an urgency.

Reviewer: ```suggestion
recommend to schedule the updates without undue haste.
```

Author: Fixed

   The most effective way for these sites to get protected is to get an
   updated version from that vendor.

> This policy is primarily a guidance and there might be exceptions to it if

Reviewer: ```suggestion
This policy is primarily guidance and there might be exceptions to it if
```

Author: Fixup pushed

they are warranted by the circumstances. For example, if a fix for a seemingly
regular issue, that was already published, is later determined to fix
a security issue, some parts of the process for handling security issues might
be skipped for that issue.

## Security release update recommendations

Our security advisories describe the affected configurations or applications.

> We always recommend to review the advisories before updating the systems

Reviewer 2: ```suggestion
We always recommend reviewing the advisories before updating the systems
```

> running with the unfixed releases.

Reviewer 2: ```suggestion
running the unfixed releases.
```

However in general it is a good idea to update the systems as soon as possible
with the security update releases that contain **Critical** or **High**
severity issue fixes.

As the **Moderate** and **Low** severity issues are either unlikely to be
exploitable and/or the impact of the successful exploit is limited, we

> recommend to schedule the updates without undue haste.

Reviewer 2: ```suggestion
recommend scheduling the updates without undue haste.
```
